packages/loader/container-loader/src/container.ts

```diff
@@ -1610,7 +1610,7 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
         }
     }
 
-    private propagateConnectionState(initialTransition: boolean) {
+    private propagateConnectionState(initialTransition: boolean, reason?: string) {
```

Comment:
Let's call it `disconnectedReason`

Reply:
(and everywhere else, like update connectionStateChanged callback declaration/implementation)

Reply:
@markfields Does the "caught up" here refer to the connected reason or is this a disconnectedReason?
I made the changes here to update the connectionStateChanged to say disconnectedReason instead of reason as you recommended but just double checking since i'm seeing this is in connectionStateHandler.ts - line 180:

<img width="905" alt="image" src="https://user-images.githubusercontent.com/49001484/193938782-62c2b27d-70dd-44f3-a124-d7c2e98d2497.png">

Reply:
It's a connected reason.  We should either delete it or rename everything back to "reason".  Taking a quick look to see what my lean is.